x/rewards/keeper/iter.go

```diff
@@ -206,6 +206,35 @@ func (k Keeper) CalculationOfRewards(
 	return newAm, nil
 }
 
+func (k Keeper) OraclePriceForRewards(ctx sdk.Context, id uint64, amt sdk.Int) (sdk.Dec, bool) {
+	asset, found := k.asset.GetAsset(ctx, id)
+	if !found {
+		return sdk.ZeroDec(), false
+	}
+
+	price, found := k.marketKeeper.GetTwa(ctx, asset.Id)
+	if !found {
+		return sdk.ZeroDec(), false
+	}
+
+	// if price is not active and twa is 0 return false
+	if !price.IsPriceActive && price.Twa == 0 {
+		return sdk.ZeroDec(), false
+	}
+	// if price is not active and DiscardedHeightDiff is not -1
+	if price.DiscardedHeightDiff != -1 {
+		priceInactiveBlockCount := ctx.BlockHeight() - price.DiscardedHeightDiff
+		// if price is inactive since 600 block and also twa is 0 return error else continue with the old price
+		if priceInactiveBlockCount >= types.DefaultAllowedBlocksForPriceInactive {
+			return sdk.ZeroDec(), false
+		}
+	}
+
+	numerator := sdk.NewDecFromInt(amt).Mul(sdk.NewDecFromInt(sdk.NewIntFromUint64(price.Twa)))
```

Comment:
`price.Twa` is in` 10^6 `format, I hope you are aware.

Reply:
Yes